models/academia/college.js

```diff
@@ -45,20 +46,28 @@ collegeSchema.pre('validate', function(next) {
 
 collegeSchema.methods.getCourse = function(courseName) {
   if (courseName instanceof RegExp) {
-    for (const course of this.courses) {
-      if (course.course.match(courseName)) {
-        return course;
-      }
-    }
+    return _.find(this.courses, (course) => {
+      return courseName.test(course.course);
+    });
   } else {
-    return utility.arrayUtil.findNeedle(
-        this.courses,
-        courseName,
-        'course',
-        true,
-    );
+    return _.find(this.courses, (course) => {
+      return course.course == courseName;
+    });
```

Comment:
is this necessary if regex can also handle this? and if this is necessary then please add a similarly `RegExp` check to `getSemester` also otherwise it looks inconsistent across similar kind of api's

Reply:
Yes, regex can handle this, it seems `else` statement block is redundant here, `getCourse` is always called with regex if I am right. Will check regarding this and modify `getCourse`, `getBranch` and `getSubject` function accordingly

Reply:
Yes, the case handled in else block was redundant, I have now modified `getCourse`, 'getBranch' and `getSubject` accordingly.